### PR TITLE
TOML: fix some printing of special characters

### LINKED ISSUE #22: TOML printing of strings escaped with `\xnn` doesn't round trip
I was playing with storing binary data in certian TOML strings, and discovered TOML.jl doesn't escape these on output in a way which can be read back in again using TOML.parse.  For example:

```julia
julia> TOML.print(Dict("x"=>"\x01"))
x = "\x01"

julia> TOML.parse(sprint(TOML.print, Dict("x"=>"\x01")))
ERROR: TOML Parser error:
none:1:7 error: invalid escape character
  x = "\x01"
        ^    
Stacktrace:
 [1] parse
   @ ./toml_parser.jl:441 [inlined]
 [2] parse(str::String)
   @ TOML /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/TOML/src/TOML.jl:71
 [3] top-level scope
   @ REPL[80]:1
```

It seems that `TOML.print` should be escaping bytes as unicode codepoints with `\uXXXX`, or `\UXXXXXXXX`, instead (https://toml.io/en/v1.0.0#string). The parser works for this.

```
julia> TOML.parse("x = \"\\u0001\"")
Dict{String, Any} with 1 entry:
  "x" => "\x01"
```

### PR DESCRIPTION
**Pull request by KristofferC**
Fixes https://github.com/JuliaLang/TOML.jl/issues/22.

### CODE REVIEW

File: stdlib/TOML/src/print.jl [hunk 1]
```diff
@@ -5,6 +5,34 @@ import Dates
...
+            Base.print(io, '\\', 'r')
+        elseif c == '"'
+            Base.print(io, '\\', '"')
+        elseif c == '\\'
+            Base.print(io, "\\", '\\')
+        elseif Base.iscntrl(c)
+            Base.print(io, "\\u")
+            Base.print(io, string(UInt32(c), base=16, pad=4))
+        else
+            Base.print(io, c)
```
Comment (stevengj) [inline stdlib/TOML/src/print.jl:31]: Should it throw an error for unrecognized escape sequences?

Comment (KristofferC) [reply stdlib/TOML/src/print.jl:31]: Hm, maybe? Could you give an example.

Comment (stevengj) [reply stdlib/TOML/src/print.jl:31]: See #22800 and #27238 for discussion and examples of analogous issues in the Julia parser and in `unescape_string`.

Comment (KristofferC) [reply stdlib/TOML/src/print.jl:31]: What I meant is more an example Julia string that should error upon writing it to a TOML file.

Comment (stevengj) [reply stdlib/TOML/src/print.jl:31]: Oh never mind. I confused escaping with unescaping here.

File: stdlib/TOML/test/print.jl [hunk 2]
```diff
@@ -60,3 +60,14 @@ end
...
+
+@testset "special characters" begin
+    s = """
+    "\U1f355 \0 \x0 \x1 \t \b" = "\U1f355 \0 \x0 \x1 \t \b"
+    "\x7f" = "\x7f"
+    """
+    @test roundtrip(s)
+
+    d = Dict("str" => string(Char(0xd800)))
+    @test_throws ErrorException TOML.print(d)
```
Comment (vtjnash) [inline stdlib/TOML/test/print.jl:72]: This appears to have introduced this defect in the test output:
```
TOML                               (41) |    31.70 |   0.67 |  2.1 |    1230.28 |  1353.48
      From worker 41:   str = "
```
```suggestion
    @test_throws ErrorException TOML.print(devnull, d)
```

**Pull request merged**